### PR TITLE
Fix bugs and implement JsonArchiveStorageTest

### PR DESCRIPTION
**Pull request by rgbpokka**

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -94,14 +97,21 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
      * The data from the sample address book will be used instead if {@code storage}'s address book is not found,
      * or an empty address book will be used instead if errors occur when reading {@code storage}'s address book.
      */
-    private ReadOnlyGuestBook initGuestBook(Storage storage) {
+    private ReadOnlyGuestBook initGuestBook(Storage storage, ReadOnlyGuestBook archive) {
```
Comment (NicolasCwy) [inline src/main/java/seedu/address/MainApp.java:100]: What is the purpose of adding archive to this? It seems to have worked previously and you don't seem to have changed any code that use this?

**Review by NicolasCwy — CHANGES_REQUESTED** [review]
Some small changes, also could you run the tests and see if all are passing?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -632,6 +632,15 @@ his/her details to the receptionist upon checking in to the hotel, hence there i
 refer to details of the archived guests in the UI.
 </div>
 
+#### Additional Notes
```
[the remark this answers is not in the document]
Comment (rgbpokka) [reply docs/UserGuide.md:635]: Corrected

File: src/main/java/seedu/address/MainApp.java [hunk 3]
```diff
@@ -4,6 +4,9 @@
 import java.nio.file.Path;
 import java.util.Optional;
 import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import org.bouncycastle.util.io.Streams;
```
[the remark this answers is not in the document]
Comment (rgbpokka) [reply src/main/java/seedu/address/MainApp.java:9]: Corrected

File: src/main/java/seedu/address/MainApp.java (hunk 1 above)
Comment (rgbpokka) [reply src/main/java/seedu/address/MainApp.java:100]: `initGuestBook` loads sample guest data if there is no pre-existing guest data. We want to make sure that the sample data loaded does not conflict with the archived data (by using the same passport numbers as those already in the archive). Hence initGuestBook takes in archive, which will be passed to `getSampleGuestBook`, where the necessary checks will be done. 

image

**Review by NicolasCwy — APPROVED** [review]
LGTM

**Pull request merged**